packages/meditrak-server/src/routes/utilities/translateQuestionDependentFields.js

```diff
@@ -0,0 +1,76 @@
+import { splitStringOn } from './split';
+
+// JSON subfields will be in the format field.subfield, e.g. 'attributes.type'
+const isJsonSubfield = (jsonFieldList, fieldKey) =>
+  jsonFieldList.find(field => fieldKey.startsWith(field));
+
+export const translateQuestionDependentNonJsonFields = (config, nonJsonFieldTranslation) => {
```

Comment:
I moved the code that translate/replace the config into 1 util file like this to reduce code duplication. However, it felt a bit awkward for me to translate both non json (eg: parent, grandparent) and json (attributes.type, attributes.parent_project) in 1 method. So I split it into 2 different methods, 1 for translating non json and 1 for translating json which I feel is a bit cleaner (although it have a little bit of code duplication)

Reply:
Good call. I would remove the `NonJson` bit from the function and variable names though. Having the other one include `Json` is enough to distinguish the two.

Reply:
Actually I would use the term `Nested` rather than `Json`...